### PR TITLE
Add airGappedChecksum field to check workload version before rollout

### LINKED ISSUE #297: in airGapped mode, ensure that the desired k8s version is used
We noticed [while developing Sylva](https://gitlab.com/sylva-projects/sylva-core/-/issues/1081) that, in airGapped mode (`spec.agentConfig.airGapped: true`), the combination of RKE2 bootstrap provider  and RK2 itself will result in using _whatever version of RKE2 is present in `/opt/rke2-artifacts`, even if this version does not match `spec.agentConfig.version`.

It creates a risk of unintendedly using an undesired version, and possibly unintendedly upgrading or downgrading, if the OS image is not selected carefully.

A safer behavior for air-gapped mode would be to abort RKE2 installation if the RKE2 version that can be found on disk does not match `spec.agentConfig.version`.

/cc @richardcase

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:
This change adds `AirGappedChecksum` field to agent config. By using this field, a user might declare expected sha256sum checksum for the air gapped manifests and prevent incomplete or different package installation on the host.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #297 
Depends on #313 for fuzzy test changes to make CI green

**Special notes for your reviewer**:

Internals of the `curl -sfL https://get.rke2.io` in the air-gapped mode already verify the contents of provided tar files based on the checksum from `sha256sum-<arch>.txt`. Therefore the `AirGappedChecksum` declares the expected checksum for this file only to connect API definition with the prepared host pre-bootstrap state.

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: bootstrap/api/v1beta1/rke2config_types.go
Comment (alexander-demicev) [inline bootstrap/api/v1beta1/rke2config_types.go:162]: can we make this field optional?

**Review by alexander-demicev — APPROVED** [review]
thanks, looks good just one small comment

**Comment by salasberryfin** [comment]
LGTM pending fixing CI checks and resolving @alexander-demicev's comment.

File: bootstrap/api/v1beta1/rke2config_types.go
Comment (Danil-Grigorev) [reply bootstrap/api/v1beta1/rke2config_types.go:157]: It was blocked on #313 for fuzzy testing, so I was only able to add generated manifests, make the field optional and backward value recovery after it got merged. This should be ready for review.

**Review by alexander-demicev — APPROVED** [review]

**Pull request merged**
